### ISSUE TITLE
[Accessibility] Narrator rectangle can't  focus on the entire context of RichTextBox control when setting  the RightToLeft property as true

### ISSUE THREAD

**Issue by Zheng-Li01** [dotnet/winforms#4754]
* .NET Core Version: 
.NET 6.0.100-preview.4.21181.20

**More info:**
1.  The issue can't reproduce on TextBox & ComboBox controls.
2.  This is not a regression issue, also can reproduce on .NET SDK 3.1 & 5.0.
3. Narrator can announce the entire context of RichTextBox contorl correctly, fixed in GH issue [3455](https://github.com/dotnet/winforms/issues/3455).

**Have you experienced this same bug with .NET Framework?:** 
 Yes 

**Problem description:**
Can't see the rectangle for RichTextBox contorl when setting the RightToLeft property as true.

**Expected behavior:**
Narrator rectangle can  focus on the entire context of RichTextBox contorl when setting the RightToLeft property as true.

**Actual behavior:**
Narrator rectangle can't  focus on the entire context of RichTextBox contorl when setting property the RightToLeft as true
![CoreNewIssue](https://user-images.githubusercontent.com/38325459/113266521-f3d7ea80-9307-11eb-8a66-b8ef02effa40.gif)

**Minimal repro:**
1. Extract the application. [WinFormsApp4.zip](https://github.com/dotnet/winforms/files/6242165/WinFormsApp4.zip).
2. Build and run it.
3. Open the Narrator tool, and press the Tab keyboard to focus the contorl one by one.
4. Observe the narrator rectangle for RichTextBox contorl which setting the RightToLeft property as true.

**Comment by vladimir-krestov** [comment]
@Zheng-Li01, please check if this is a duplicate of #3640. If so, please close one of them.

**Comment by Marie-Shi** [comment]
@vladimir-krestov Maybe those two issues are not same. 
For [4754](https://github.com/dotnet/winforms/issues/4754), there some contents in RichTextBox, when we Tab to RichTextBox, first there is no Narrator's blue cursor(Narrator rectangle can't focus on the entire context of RichTextBox), then we press Left arrow key, it will focus on single character
![4754](https://user-images.githubusercontent.com/45864017/137076595-9e26b850-8ab6-4625-ae4f-940ce45c9a28.png)
.
![4754](https://user-images.githubusercontent.com/45864017/137076601-687ca1ff-94f1-42b5-a802-0b590512b980.gif)
For [3640](https://github.com/dotnet/winforms/issues/3640), it is an empty RichTextBox, original issue is about Narrator's blue cursor cannot move when entering characters into the RichTextBox/ComboBox in RTL environment. But verified it in latest 7.0.100-alpha.1.21512.13 and 6.0.100-rtm.21508.19, Narrator's blue cursor can move. But for RichTextBox, it cannot focus on the new   input character, just focus on left.
![3640](https://user-images.githubusercontent.com/45864017/137077552-892e8595-ec40-413a-ade3-e93a9dd2053d.png)
![3640](https://user-images.githubusercontent.com/45864017/137077565-21263ce1-d6fd-46bd-8991-49458e5ca297.gif)

**Comment by vladimir-krestov** [comment]
Any known ways don't affect Narrator to focus on the RTL RichTextBox, this control doesn't support UIA, so this bug will be fixed after implementing of UIA for RichTextBox, but it may be time consuming task.

**Comment by dmitrii-drobotov** [comment]
After investigating this issue, I didn't find a good way to resolve it for the RichTextBox.

For the regular TextBox it works by selecting all of its text, which then determines the highlight rectangle. So one way of fixing it could be to also select all text for RichTextBox when it gets focused. But after implementing it I found that RichTextBox has a side effect of scrolling the content to the end of selection:

https://user-images.githubusercontent.com/102954094/177601045-2bcad4e2-659f-49f4-8394-145f36ae5ae2.mp4

It does solve the issue but the scrolling seems to be an undesired behavior and could be a breaking change. Therefore I don't thinks this is a good solution. 

Another fix that I found is raising an additional focus changed event (`AccessibilityObject.RaiseAutomationEvent(UiaCore.UIA.AutomationFocusChangedEventId);`):

https://user-images.githubusercontent.com/102954094/177605330-a0e78ab5-8903-4a1a-a2c2-4bec3b14d16a.mp4

But it also has limitations:
* Doesn't add a rectangle when opening a form if RichTextBox is a first control on a form, only after changing the focus back 
* Not very stable when done fast and adds a "flickering" in some cases when RichTextBox already had a highlight rectangle

https://user-images.githubusercontent.com/102954094/177601160-5337e67a-ae48-49a9-885c-d56bfefa9334.mp4

* RichTextBox also has a text notification to announce its content, and when focus notification is raised along with it, it makes Narrator skip RichTextBox's name and type announcements, probably due to a contention problem

I couldn't find a way to bypass any of above issues with focus notification, so I also don't think this would be a way to go with.

We might have more flexibility on solving this if we had our own implementation of RichTextBox's accessibility object. But as I found when looking into https://github.com/dotnet/winforms/issues/3640 it would take a lot of effort, which probably isn't worth it.

Considering all of the above I would say that this issue is out of our scope along with #3640 and we should leave it as is for now. 

CC @merriemcgaw @vladimir-krestov

**Comment by merriemcgaw** [comment]
@dmitrii-drobotov , I agree. I can't see the RTL scenarios as being critical for WinForms in particular. I'll close these two.
